packages/components/src/Chips/InternalChipDismissible/InternalChipDismissibleInput.tsx

```diff
@@ -107,24 +100,36 @@ export function InternalChipDismissibleInput(props: ChipDismissibleInputProps) {
             className={styles.menuList}
             data-testid="chip-menu"
           >
-            {allOptions.map((option, i) => (
-              <button
-                key={option.value}
-                role="option"
-                type="button"
-                id={generateDescendantId(i)}
-                className={classNames(styles.menuListOption, {
-                  [styles.activeOption]: activeIndex === i,
-                })}
-                onClick={() => handleSelectOption(option)}
-                onMouseEnter={handleSetActiveOnMouseOver(i)}
-                onMouseDown={handleCancelBlur}
-                onMouseUp={handleEnableBlur}
-              >
-                <span aria-hidden>{option.prefix}</span>
-                <Text>{option.label}</Text>
-              </button>
-            ))}
+            {props.customRenderMenu
+              ? props.customRenderMenu(allOptions, {
+                  generateDescendantId,
+                  handleSelectOption,
+                  handleSetActiveOnMouseOver,
+                  handleCancelBlur,
+                  handleEnableBlur,
+                  handleShowInput,
+                  activeIndex,
+                  menuListOptionStyles: styles.menuListOption,
+                  activeOptionStyles: styles.activeOption,
+                })
```

Comment:
not the most elegant that each and every one of these has to be implemented, but what this does enable is that someone can drop their own button at the end of these items

however if you want the default one there's not really a great way to get that with this setup because we opted to skip out on all the search behavior and that's tied to the internal search. we'd need another way to have both.